doc: add note about AsyncResource for Worker pooling (#28023)
nodejs/node: doc/api/worker_threads.md

The above example spawns a Worker thread for each `parse()` call. In actual
practice, use a pool of Workers instead for these kinds of tasks. Otherwise, the
overhead of creating Workers would likely exceed their benefit.

> When implementing a worker pool, it is strongly recommended to use the

Reviewer: Optional suggestion:

```suggestion
When implementing a worker pool, use the
```

> [`AsyncResource`][] API to inform diagnostic tools (for e.g. asynchronous stack

Reviewer: Typo: `for e.g.` should be either `for example` or `e.g.`, although I would go with `such as`:

```suggestion
[`AsyncResource`][] API to inform diagnostic tools (such as asynchronous stack
```

Author: @trott I don’t think `for example` or just `e.g.` would work here, as asynchronous stack traces aren’t diagnostic tools by themselves, but rather provided by them. I’ve changed this to `e.g. in order to provide asynchronous stack traces`, maybe that works for you?

Reviewer: Yes, that's great.

Reviewer: (Total ignorable nit: Maybe drop "in order"?)
